src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+export type Shape = 'triangle' | 'circle' | 'rectangle';
+export type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+function checkOnError(errorMessage: string, ...args: number[]): void {
+  if (args.some((arg: number) => arg <= 0)) {
+    throw new Error(errorMessage);
+  }
+}
+
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    checkOnError('Triangle side length must be greater than zero', a, b, c);
+
+    if (a + b <= c || c + b <= a || c + a <= b) {
+      throw new Error(`
+        throws an error: sides ${a}, ${b} and ${c} can’t form a triangle
+      `);
```

Comment:
The error message should be more descriptive. Instead of using a template literal with newlines, provide a single-line error message that includes the condition that caused the error.